### PR TITLE
HWKAPM-533 | Add http method from Span to derived information objects

### PR DESCRIPTION
**Pull request by pavolloffay**
https://issues.jboss.org/browse/HWKAPM-533

### CODE REVIEW

**Comment by pavolloffay** [comment]
@objectiser could you please look when you have some time?

File: processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/SpanDeriverUtil.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private SpanDeriverUtil() {}
+
+    /**
+     * Derives http operation from span.
+     *
+     * @param span the Span
+     * @return operation (e.g. HTTP GET/POST)
+     */
+    public static String deriveHttpOperation(Span span) {
+        List<HttpCodesUtil.HttpCode> httpStatusCodes = HttpCodesUtil.getHttpStatusCodes(span.getBinaryAnnotations());
```
Comment (objectiser) [inline processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/SpanDeriverUtil.java]: This method should be made more generic - i.e. 'deriveOperation(span)' - and only internally do we check for HTTP as being one specific type from which we derive the operation name.

File: processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/SpanDeriverUtil.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if (httpStatusCodes.size() > 0) {
+            return span.getName();
+        }
```
Comment (objectiser) [inline processors/zipkin/src/main/java/org/hawkular/apm/processor/zipkin/SpanDeriverUtil.java]: If we determine it is a http span, then I think we should check that the name is one of the HTTP methods, e.g. GET, POST, etc. before returning it. If it isn't then we will have to have another strategy.

**Comment by objectiser** [comment]
Review finished - just a couple of minor changes.

**Comment by pavolloffay** [comment]
Updated version is here.

**Comment by objectiser** [comment]
@pavolloffay LGTM :+1:

**Pull request merged**